iphone/Classes/TiUIListView.m

```diff
@@ -2041,74 +2030,52 @@ - (void)searchBarCancelButtonClicked:(UISearchBar *) searchBar
     [self buildResultsForSearchText];
 }
 
-#pragma mark - UISearchDisplayDelegate Methods
+#pragma mark - UISearchControllerDelegate
+
+- (void)willDismissSearchController:(UISearchController *)searchController
+{
+    self.searchString = @"";
+    [self buildResultsForSearchText];
+    if (_searchWrapper != nil) {
+        CGFloat rowWidth = floorf([self computeRowWidth:_tableView]);
+        if (rowWidth > 0) {
+            CGFloat right = _tableView.bounds.size.width - rowWidth;
+#ifndef TI_USE_AUTOLAYOUT
+            [_searchWrapper layoutProperties]->right = TiDimensionDip(right);
+#endif
+             [_searchWrapper refreshView:nil];
+        }
+    }
+    //IOS7 DP3. TableView seems to be adding the searchView to
+    //tableView. Bug on IOS7?
+    [self clearSearchController:self];
+    [_tableView reloadData];
+}
 
-- (void)searchDisplayControllerWillBeginSearch:(UISearchDisplayController *)controller
+- (void)presentSearchController:(UISearchController *)controller
 {
+    id dimBgValue = [[self  proxy] valueForKey:@"dimsBackground"];
+    searchController.dimsBackgroundDuringPresentation = [TiUtils boolValue:dimBgValue def:NO];
```

Comment:
This property is of no use, until searchResultController is passed in UISearchViewController. So removed.